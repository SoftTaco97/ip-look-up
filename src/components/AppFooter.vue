<template>
  <v-container fluid>
    <v-row justify="center">
      <h3>Jared Martinez | {{ new Date().getFullYear() }} &copy; </h3>
    </v-row>
  </v-container>
</template>
