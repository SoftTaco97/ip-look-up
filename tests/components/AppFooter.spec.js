import createComponentMounter from '../componentMounter';
import AppFooter from '../../src/components/AppFooter.vue';

describe('src/components/AppFooter.vue unit tests', () => {
  const mountComponent = createComponentMounter(AppFooter);
  let wrapper;

  beforeEach(() => {
    wrapper = mountComponent();
  });

  afterEach(() => wrapper.destroy());

  describe('rendering unit tests', () => {
    it('Should match the snapshot', () => {
      // ! - Deep mount for snapshots
      wrapper = mountComponent({}, false);
      expect(wrapper).toMatchSnapshot();
    });
    it('Should the proper information', () => {
      const html = wrapper.html();
      expect(html).toEqual(expect.stringContaining('Jared Martinez'));
      expect(html).toEqual(expect.stringContaining(new Date().getFullYear().toString()));
    });
  });
});
