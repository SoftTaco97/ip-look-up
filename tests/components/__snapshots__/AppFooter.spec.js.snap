// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/AppFooter.vue unit tests rendering unit tests Should match the snapshot 1`] = `
<div class="container container--fluid">
  <div class="row justify-center">
    <h3>Jared Martinez | 2020 Â© </h3>
  </div>
</div>
`;
